src/probes/bert_probe.c

```diff
@@ -0,0 +1,145 @@
+                goto out1;
+        }
+
+        *payload = memset(*payload, 0, MAX_PAYLOAD_SIZE);
+        ret = 1;
+
+out1:
+        return ret;
+}
+
+int main(int argc, char **argv)
+{
+        struct telem_ref *tm_handle = NULL;
+        char *classification = (char *) telem_record_class;
+        char *payload = NULL;
+
+        // Following vars are for arg parsing.
+        int c;
+        char *cfile = NULL;
+        int opt_index = 0;
+        int ret = 0;
+
+        struct option opts[] = {
+                { "cfile", 1, NULL, 'f' },
+                { "help", 0, NULL, 'h' },
+                { "version", 0, NULL, 'V' },
+                { NULL, 0, NULL, 0 }
+        };
+
+        while ((c = getopt_long(argc, argv, "f:hHV", opts, &opt_index)) != -1) {
+                switch (c) {
+                        case 'f':
+                                if (asprintf(&cfile, "%s", (char *) optarg) < 0) {
+                                        exit(EXIT_FAILURE);
+                                }
+
+                                struct stat buf;
+                                ret = stat(cfile, &buf);
+
+                                /* check if the file exists and is a regular file */
+                                if (ret == -1 || !S_ISREG(buf.st_mode)) {
+                                        exit(EXIT_FAILURE);
+                                }
+
+                                tm_set_config_file(cfile);
+                                break;
+                        case 'h':
+                                print_usage(argv[0]);
+                                exit(EXIT_SUCCESS);
+                        case 'V':
+                                printf(PACKAGE_VERSION "\n");
+                                exit(EXIT_SUCCESS);
+                        case '?':
+                                exit(EXIT_FAILURE);
+                }
+        }
+
+        if (!(ret = allocate_payload_buffer(&payload))) {
+                printf("Unable to allocate more memory.\n");
+                goto fail;
+        }
+
+        if(!(ret = nc_b64enc_filename(bert_record_file, payload, MAX_PAYLOAD_SIZE))) {
+                printf("Failed to read payload from: %s\n", bert_record_file);
+                goto fail;
+        }
+
+        if ((ret = tm_create_record(&tm_handle, severity, classification,
+                                    payload_version)) < 0) {
+                printf("Failed to create record: %s\n", strerror(-ret));
+                goto fail;
+        }
+
+        if ((ret = tm_set_payload(tm_handle, payload)) < 0) {
+                printf("Failed to set record payload: %s\n", strerror(-ret));
+                goto fail;
+        }
+
+        free(payload);
+
+        if ((ret = tm_send_record(tm_handle)) < 0) {
+                printf("Failed to send record to daemon: %s\n", strerror(-ret));
+                goto fail;
+        }
+
+        printf("Successfully sent record to daemon.\n");
```

Comment:
I would remove this line. It's intended for interactive use, but I doubt bertprobe will be often used interactively.